Library/Koala/KoalaTrackingClient.swift

```diff
@@ -99,59 +99,59 @@ public final class KoalaTrackingClient: TrackingClientType {
     }
   }
 
-  private func load() {
-    dispatch_async(self.queue) {
+  fileprivate func load() {
+    self.queue.async {
       guard
-        let file = self.fileName()
-        where NSFileManager.defaultManager().fileExistsAtPath(file),
-        let buffer = NSKeyedUnarchiver.unarchiveObjectWithFile(file) as? [[String: AnyObject]]
+        let file = self.fileName(), FileManager.default.fileExists(atPath: file),
+        let buffer = NSKeyedUnarchiver.unarchiveObject(withFile: file) as? [[String:Any]]
         else { return }
 
       self.buffer = buffer + self.buffer
 
-      _ = try? NSFileManager.defaultManager().removeItemAtPath(file)
+      _ = try? FileManager.default.removeItem(atPath: file)
     }
   }
 
-  private static func base64Payload(payload: [AnyObject]) -> String? {
-    return (try? NSJSONSerialization.dataWithJSONObject(payload, options: []))
-      .map { $0.base64EncodedStringWithOptions([]) }
+  fileprivate static func base64Payload(_ payload: [Any]) -> String? {
+    return (try? JSONSerialization.data(withJSONObject: payload, options: []))
+      .map { $0.base64EncodedString(options: []) }
   }
 
-  private func koalaURL(dataString: String) -> NSURL? {
+  fileprivate func koalaURL(_ dataString: String) -> URL? {
     #if DEBUG
       if dataString.characters.count >= 10_000 {
         print("[Koala Error] Base64 payload is longer than 10,000 characters.")
       }
     #endif
-    return NSURL(string: "\(self.endpoint.base)?data=\(dataString)")
+    return URL(string: "\(self.endpoint.base)?data=\(dataString)")
   }
 
-  private static func koalaRequest(url: NSURL) -> NSURLRequest {
-    let request = NSMutableURLRequest(URL: url)
-    request.HTTPMethod = "POST"
+  fileprivate static func koalaRequest(_ url: URL) -> URLRequest {
+    var request = URLRequest(url: url)
+    request.httpMethod = "POST"
     return request
   }
 
-  private func synchronousKoalaResult(request: NSURLRequest) -> Result<NSHTTPURLResponse, NSError>? {
-    var result: Result<NSHTTPURLResponse, NSError>?
-    let semaphore = dispatch_semaphore_create(0)
+  fileprivate func synchronousKoalaResult(_ request: URLRequest) -> Result<HTTPURLResponse, NSError>? {
+    var result: Result<HTTPURLResponse, NSError>?
+    let semaphore = DispatchSemaphore(value: 0)
 
-    self.URLSession.dataTaskWithRequest(request) { _, response, err in
-      defer { dispatch_semaphore_signal(semaphore) }
+    self.URLSession.dataTask(with: request) { _, response, err in
+      defer { semaphore.signal() }
 
-      if let httpResponse = response as? NSHTTPURLResponse {
+      if let httpResponse = response as? HTTPURLResponse {
         #if DEBUG
           NSLog("[Koala Status Code]: \(httpResponse.statusCode)")
         #endif
         result = Result(value: httpResponse)
       }
 
-      if let err = err {
+      // FIXME: is this cast safe? is it possible to get a non-NSError here?
+      if let err = err as? NSError {
```

Comment:
actually nvm that... it's still strange.

`URLSession.dataTask(with:)` can return a generic `Error` type. if i wanna package that up into a `Result` i need to know what type of error it is, and so would need to cast, and hence could potentially miss out on some errors :/

Since we don't actually care what the error is, just that it either did or did not error, i'm gonna handle this in a much simpler way here 08a9e38